packages/react-three-rapier/src/utils-collider.ts

```diff
@@ -82,9 +82,44 @@ export const immutableColliderOptions: ImmutableColliderOptions = [
 ];
 
 type MutableColliderOptions = {
-  [key in keyof ColliderProps]: (collider: Collider, value: any) => void;
+  [key in keyof ColliderProps]: (
+    collider: Collider,
+    value: Exclude<ColliderProps[key], undefined>,
+    options: ColliderProps
+  ) => void;
 };
 
+const defaultMassRotation = vector3ToQuaternion(new Vector3(0, 0, 0)).clone();
+
+type MassPropertiesType = 'mass' | 'centerOfMass' | 'principalAngularInertia' | 'density';
+const fixMass = (collider: Collider, options: Pick<ColliderProps, MassPropertiesType>) => {
```

Comment:
This is not necessary anymore.

We only need to set all of these when creating the colliderDesc, as there is no alternative. Now the colliderDesc is completely separate from setting the options.

The `mass * 0.2` was a random estimation by myself, and not accurate from what I know.

Setting the mass, or density, after the collider is created works as you'd expect. The `principalAngularInertia` is calculated by Rapier automatically.  Making all this much simpler.

Instead I'd allow people to set `massProperties` if they want to do their own calculations, but the use-case it pretty narrow.

Reply:
Makes sense.

So to be clear, shall we have these proprieties:
`mass` `density` and `massProperties`, and in `massProperties` they specify the whole struct?

My main concern is that I want to set centerOfMass.

I also want to be able to set both density and center of mass... I will think about that.

Reply:
Yeah, `centerOfMass` should probably not be a prop, since it's only able to be set via `setMassProperties`.

I think allowing people to set `mass`, `density`, and `massProperties` independently makes most sense.
People just gotta be aware that setting one will overwrite the properties of the other.

Reply:
@wiledal can you please take a look? I have updated the PR :)